007-design-docs.md

```diff
@@ -0,0 +1,124 @@
+An example of this might be something like:
+```
+Given a user deletes the CA certificate
+When the operator reconciles
+Then the operator will replace it
+```
+or
+```
+Given a KafkaRebalance proposal is approved
+When the KafkaRebalance enters the Ready state
+Then the cluster will be rebalanced to 'X' state
+When the KafkaRebalance is refreshed
+Then the KafkaRebalance has no proposal
+And the KafkaRebalance will be in the Ready state
+```
+
+
+### How to write a design document
+
+The design documents would live in the related repository in the Strimzi Organization and would match the current code state, so that code changes could be audited with their corresponding design doc changes.
+
+A design document would be split into the following parts:
+
+**Title**
+
+The title of the design doc, normally this would pertain to a single component or one larger aspect of the whole of Strimzi. New pages would be discussed and decided on as part of a proposal where needed.
+
+**Overview and Context**
+
+A high level summary that any user of Strimzi should be able understand and use to decide if it’s useful for them to read the rest of the doc.
+A description of the problem the program/code tackles, why the program/code is necessary, what people need to know to assess the success of this design.
+Optionally include a high level description describing the current implementation as well as example flow(s) to illustrate how users interact with this system and/or how data flow through it.
+
+**Goals and Non-Goals**
+
+The Goals section should describe the user-driven impact of the code and specify how to measure success using metrics, normally that would be tested.
+Non-Goals are equally important to describe which problems you won’t be fixing so everyone is on the same page.
+
+**Technical Architecture section**
+
+Walk through the flows in great detail of the implementation. Feel free to include diagrams where relevant and sub-sections for edge-cases and implementation details.
+This section would contain test scenarios in the form of behaviour driven tests in the [Gherkin Syntax](https://cucumber.io/docs/bdd/better-gherkin/) if applicable.
+
+
+## Summary of changes
+
+In an effort to improve where we are, the above guidelines are suggestions for future contributors but no enforced or mandated. This means that we will not place the entire burden on the first person to come along who wants to make a change in a given area. 
+Instead with larger coontributions and proposed changes we would request contributors to add to whatever framework of design documentation already exists.
```

Comment:
I'm afraid I do not understand what does this mean. Is this trying to say that given missing design docs, you should not need to write them from zero when fixing some smaller bugs and issues?